org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/StatusWidget.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2015, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xtext.ui.wizard.project;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.dialogs.IMessageProvider;
+import org.eclipse.jface.resource.JFaceResources;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.xtext.xbase.lib.Procedures.Procedure0;
+
+/**
+ * @author dhuebner - Initial contribution and API
+ */
+public class StatusWidget extends Composite {
+
+	private Link link;
+
+	private Label imageLabel;
+
+	private Procedure0 quickFix = () -> {};
+
+	private int severity = IMessageProvider.NONE;
+
+	public StatusWidget(Composite parent, int style) {
+		super(parent, style);
+		createControls();
+	}
+
+	protected void createControls() {
+		setLayout(new GridLayout(2, false));
+
+		imageLabel = new Label(this, SWT.NONE);
+		imageLabel.setText("   ");
+		GridData imgLabelLayoutData = new GridData(GridData.VERTICAL_ALIGN_BEGINNING);
+		imgLabelLayoutData.heightHint = 15;
+		imageLabel.setLayoutData(imgLabelLayoutData);
+
+		link = new Link(this, SWT.NONE);
+		GridData linkLayoutData = new GridData(GridData.FILL_HORIZONTAL);
+		linkLayoutData.heightHint = 30;
+		link.setLayoutData(linkLayoutData);
+		link.setFont(getFont());
+		link.setText("\n\n\n");
+		link.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				super.widgetSelected(e);
+				quickFix.apply();
+			}
+		});
+	}
+
+	public Procedure0 clearStatus() {
+		return setStatus(IMessageProvider.NONE, "\n\n\n", () -> {}, () -> {});
+	}
+
+	public Procedure0 setStatus(int severity, String text, Procedure0 quickFix, Procedure0 callback) {
+		this.severity = severity;
+		setVisible(severity != IMessageProvider.NONE);
+		imageLabel.setImage(imageFor(severity));
+		link.setText(text);
+		Matcher matcher = Pattern.compile("<a>(.*)</a>").matcher(text.trim());
```

Comment:
Not sure how often this has to run but precompiling the regex might be a good idea